src/main/java/org/htmlunit/javascript/host/URLSearchParams.java

```diff
@@ -137,26 +143,153 @@ public URLSearchParams() {
      */
     @JsxConstructor
     public URLSearchParams(final Object params) {
-        // TODO: Pass in a sequence
-        // new URLSearchParams([["foo", 1],["bar", 2]]);
-
-        // TODO: Pass in a record
-        // new URLSearchParams({"foo" : 1 , "bar" : 2});
-
         url_ = new URL("http://www.htmlunit.org", "");
 
         if (params == null || Undefined.isUndefined(params)) {
             return;
         }
 
         try {
-            url_.setSearch(splitQuery(Context.toString(params)));
+            url_.setSearch(splitQuery(createParamString(params)));
+        }
+        catch (final EcmaError e) {
+            throw ScriptRuntime.typeError("Failed to construct 'URLSearchParams': " + e.getErrorMessage());
         }
         catch (final MalformedURLException e) {
             LOG.error(e.getMessage(), e);
         }
     }
 
+    private static String createParamString(final Object params) {
```

Comment:
i think this makes more sense

```suggestion
    private static List<NameValuePair> resolveParams(final Object params) {
```